src/pfs/pfs.proto

```diff
@@ -114,8 +114,10 @@ message CommitOrigin {
 // protos)
 message Commit {
```

Comment:
This approach is a bit concerning to me, but I see why you went with it. This allows us to only use the repo & id internally, while not breaking the public API which can specify commits in terms of repo & branch that will resolve to the head. This may be the way to go for now, but I would make a clear note such that we can reconsider this for the next major release. I think request protos that support specifying branches in place of commits should operate through a oneof with a commit or branch rather than a commit with a branch field and empty id.

Reply:
This is a bit of a nitpicky thing, but I think we should order the fields as repo, id, branch in the proto (of course not changing the field numbers though). That lines up a bit better with how we are thinking about things now in that a repo and id is expected to identify a commit, while the branch field is just there for compatibility. It looks like you reorganized the constructor for a commit to be like this anyways.